### PR TITLE
Add draw.io plugin

### PR DESCRIPTION
**Pull request by ItayZiv**
Adds the draw.io plugin from modelmat https://github.com/Modelmat/sphinxcontrib-drawio.
Allows using native x.drawio files and generating images from them.

Still WIP because there are a few CI related kinks to work out, and waiting on width/height to get implemented since otherwise, images look low-res

Fixes: https://github.com/wpilibsuite/frc-docs/issues/237

### CODE REVIEW

File: source/docs/contributing/images/example.drawio [hunk 1]
```diff
@@ -0,0 +1 @@
+<mxfile host="Electron" modified="2020-04-04T23:22:47.383Z" agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) draw.io/12.6.5 Chrome/80.0.3987.86 Electron/8.0.0 Safari/537.36" etag="4e_k7_sMNY4Ddl0NmqiL" version="12.6.5" type="device"><diagram id="tHqYu6qWHXjSS9A0nAWg" name="Page-1">jZNLb4JAEIB/DccmPFLLVdDaJrYXk9brlh1h4z7osgj213coA0iNiSd2vnk/8KJUtRvLyuLNcJBe6PPWi1ZeGAZ+HOOnI+eexLHfg9wKTkYT2IkfGDyJ1oJDNTN0xkgnyjnMjNaQuRlj1ppmbnYwcp61ZDlcgV3G5DX9FNwV1MWjP/EXEHkxZA580ig2GBOoCsZNc4GitRel1hjXv1SbguyGN8yl93u+oR0Ls6DdPQ7JR5O+f9X71/220tvlQnwfNw8U5cRkTQ2vW6ZK7D70V4Lllimq3p2HkThoMWFSOCURBPisnDVHSI00Fok2Gi2Tg5DyH2JS5BrFDEsG5MkJrBM47CUplOC8S5M0hXCwK1nW5WzwtJBZU2sOXTc+SlQ4BoD25kSCcc54oGAUOHtGE3III1oN3Wb4RHIzbTpaECsutjzulNF15WPsaQH4oB0M4rTrP93FHxOtfwE=</diagram></mxfile>
```
Comment (Daltz333) [inline source/docs/contributing/images/example.drawio:1]: Put this in a images/source/ subdirectory.

Comment (ItayZiv) [reply source/docs/contributing/images/example.drawio:1]: Yea this is still unfinished, was just trying to test something on CI, I will wait to solve the other issues, not sure if the example is actually necessary

File: azure-pipelines.yml [hunk 2]
```diff
@@ -32,7 +32,18 @@ jobs:
     - script: sudo apt-get install -qy --force-yes graphviz
       displayName: 'Install GraphViz'
 
-    - script: make html
+    - script: |
+        wget -O drawio.deb https://github.com/jgraph/drawio-desktop/releases/download/v12.6.5/draw.io-amd64-12.6.5.deb
+        sudo apt-get install libasound2 ./drawio.deb
```
Comment (Modelmat) [inline azure-pipelines.yml:37]:
```suggestion
        sudo apt-get install -qy --force-yes libasound2 ./drawio.deb
```
this is what the other commands do

File: azure-pipelines.yml [hunk 3]
```diff
@@ -32,7 +32,18 @@ jobs:
...
+      displayName: 'Install draw.io'
+
+    - script: sudo apt-get install xvfb
```
Comment (Modelmat) [inline azure-pipelines.yml:40]:
```suggestion
        sudo apt-get install -qy --force-yes xvfb
```
this is what the other commands do

File: source/conf.py [hunk 4]
```diff
@@ -13,6 +13,7 @@
 # import os
 # import sys
 # sys.path.insert(0, os.path.abspath('.'))
+import os
```
Comment (Modelmat) [inline source/conf.py:16]: why is this imported

Comment (ItayZiv) [reply source/conf.py:16]: leftover from when the env variable was checked to run headless, will be needed if we need to modify drawio_headless ourselves

File: azure-pipelines.yml (hunk 2 above)
Comment (ItayZiv) [reply azure-pipelines.yml:37]: yes, that was the plan I just didn't get to that cuz I prioritized getting stuff working first

File: azure-pipelines.yml (hunk 3 above)
Comment (ItayZiv) [reply azure-pipelines.yml:40]: same as other one

File: source/docs/contributing/style-guide.rst [hunk 5]
```diff
@@ -168,6 +168,15 @@ Diagrams should be created with ``1`` new line separating content and directive.
 
 .. drawio:: images/example.drawio
```
Comment (Daltz333) [inline source/docs/contributing/style-guide.rst:169]: Should use 
```suggestion
.. drawio:: diagrams/example.drawio
```

**Comment by Daltz333** [comment]
This repository will need to include a modification in conf.py to install the required packages on RTD, as these are non-standard dependencies.

Please use the READTHEDOCS environment variable to ensure this only runs on RTD.

https://raspberrypi.stackexchange.com/questions/73940/install-a-packages-using-python-script

**Comment by Modelmat** [comment]
Dalton, if you're talking about the git:// install of the dependency, that's a temporary fix because I haven't released the next version of the extension - I was unsure whether PDFs would work properly.

**Comment by AustinShalit** [comment]
It looks like we are going to go with the SVG option. Can this be closed?

**Comment by ItayZiv** [comment]
Yup, closing now.
Gonna leave this hear for future reference:
drawio supports having `.svg` files which have the drawio XML (normally stored as `.drawio`) stored in the metadata, meaning the file is both a svg export and the source file.
Probably gonna be whats used moving forward/

**Pull request closed without merging**